src/client/Microsoft.Identity.Client/WsTrust/MexDocument.cs

```diff
@@ -225,6 +228,26 @@ private void SetPolicyEndpointAddresses(XContainer mexDocument)
             }
         }
 
+        private IEnumerable<XElement> FindElements(XContainer mexDocument, XNamespace xNamespace, string element)
+        {
+            IEnumerable<XElement> bindingElements = mexDocument.Elements().First().Elements(xNamespace + element);
+
+            //Unable to find metadata information in root node. Attempting to search document.
+            if (bindingElements.Count() == 0)
```

Comment:
I'm not familiar with XML parsing, so just a newbie question - can bindingElements be null? Probably not..

Reply:
bindingElemts.Any() has better perf than .Count() == 0

Reply:
@bgavrilMS not entirely sure but now that we allow custom federation data ill add a null check.